### PR TITLE
[soci] fixed SOCI_CXX11 flag

### PR DESCRIPTION
**Pull request by MBoldyrev**
**Describe the pull request**

- What does your PR fix? Fixes issue #

the flag was renamed in 02a5cae2c5640f2147ef7399497f8b444c40d9bf, which is ancestor of 3742c894ab8ba5fd51b6a156980e8b34db0f3063 that is used in vcpkg.

- Which triplets are supported/not supported? Have you updated the CI baseline?

All. No.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

I hope so.

### CODE REVIEW

File: ports/soci/portfile.cmake [hunk 1]
```diff
@@ -31,7 +31,7 @@ vcpkg_configure_cmake(
     PREFER_NINJA
```
Comment (NancyLi1013) [inline ports/soci/portfile.cmake]: `include(vcpkg_common_functions)` is no longer needed.
Could you please remove it?

**Comment by NancyLi1013** [comment]
Hi @MBoldyrev thanks for this PR.
Could you please also bump the version as `4.0.0-1` in [CONTROL ](https://github.com/microsoft/vcpkg/blob/master/ports/soci/CONTROL)file?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/soci/portfile.cmake (hunk 1 above)
Comment (MBoldyrev) [reply ports/soci/portfile.cmake]: did it in a separate commit, because it seems out of the original scope.

**Comment by NancyLi1013** [comment]
It seems that there is something wrong with Linux platform on CI system.
I will re run it once linux pipeline works normally.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
@MBoldyrev 
LGTM. Thanks for your PR.

Note: All features have been tested pass with the following triplets:
- x86-windows
- x64-windows-static
- x64-linux

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR!

**Pull request merged**